coriolisclient/tests/cli/test_shell.py

```diff
@@ -0,0 +1,476 @@
+# Copyright 2024 Cloudbase Solutions Srl
+# All Rights Reserved.
+
+import ddt
+import logging
+import os
+from unittest import mock
+
+from cliff import app
+from cliff import commandmanager
+from keystoneauth1 import loading
+from keystoneauth1 import session
+
+from coriolisclient.cli import shell
+from coriolisclient import client
+from coriolisclient import exceptions
+from coriolisclient.tests import test_base
+
+
+class CustomMock(mock.MagicMock):
+    def __getattr__(self, name):
+        return None
+
+
+@ddt.ddt
+class CoriolisTestCase(test_base.CoriolisBaseTestCase):
+    """Test suite for the Coriolis Shell App."""
+
+    def setUp(self):
+        super(CoriolisTestCase, self).setUp()
+        self.coriolis = shell.Coriolis()
+
+    @ddt.data(
+        {
+            "args": {
+                "os_project_id": "mock_os_project_id",
+                "os_project_name": "mock_os_project_name",
+                "os_project_domain_name": "mock_os_project_domain_name",
+                "os_project_domain_id": "mock_os_project_domain_id",
+                "os_tenant_id": "mock_os_tenant_id",
+                "os_tenant_name": "mock_os_tenant_name"
+            },
+            "api_version": '3',
+            "raise_exc": False,
+            "expected_result": True
+        },
+        {
+            "args": {
+                "os_project_id": "mock_os_project_id",
+                "os_project_name": "mock_os_project_name",
+                "os_project_domain_name": "mock_os_project_domain_name",
+                "os_project_domain_id": "mock_os_project_domain_id",
+            },
+            "api_version": '3',
+            "raise_exc": False,
+            "expected_result": True
+        },
+        {
+            "args": {
+                "os_tenant_id": "mock_os_tenant_id",
+                "os_tenant_name": "mock_os_tenant_name"
+            },
+            "api_version": '2',
+            "raise_exc": False,
+            "expected_result": True
+        },
+        {
+            "raise_exc": False,
+            "expected_result": False
+        },
+        {
+            "api_version": '2',
+            "raise_exc": True,
+        }
+    )
+    def test_check_auth_arguments(self, data):
+        args = CustomMock()
+        for key, value in data.get("args", {}).items():
+            setattr(args, key, value)
+        expected_result = data.get("expected_result")
+
+        if expected_result is not None:
+            result = self.coriolis.check_auth_arguments(
+                args,
+                api_version=data.get("api_version", None),
+                raise_exc=data.get("raise_exc", False)
+            )
+
+            self.assertEqual(
+                expected_result,
+                result
+            )
+        else:
+            self.assertRaises(
+                exceptions.CoriolisException,
+                self.coriolis.check_auth_arguments,
+                args,
+                data.get("api_version", None),
```

Comment:
Please add the keyword to argument:
```suggestion
                api_version=data.get("api_version", None),
```